stubs/FactoryBuilder.stub

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Illuminate\Database\Eloquent;
+
+/**
+ * @template TModel
+ * @template TAmount
+ */
+class FactoryBuilder
+{
+    /**
+     * Create a collection of models and persist them to the database.
+     *
+     * @param  array<string, mixed>  $attributes
+     * @phpstan-return Collection<TModel>|TModel
```

Comment:
does phpstan have support for template conditionals? I solved for these specific types over at psalm with the following https://github.com/mr-feek/psalm-plugin-laravel/pull/1/files#diff-3a21265d8cec9058a791ab9d9a40cb5aR39

Reply:
Does not seem to be supported in phpstan https://github.com/phpstan/phpstan/issues/3853